lib/runtime/formatter.js

```diff
@@ -179,23 +213,30 @@ module.exports = class Formatter extends FormatUtils {
         }), hint);
     }
 
-    _postprocess(formatted, hint) {
+    _normalize(formatted) {
         if (typeof formatted === 'string')
-            return formatted;
+            return [formatted];
         if (formatted === null)
-            return '';
+            return [];
         if (typeof formatted === 'object' &&
             formatted.type === 'text')
-            return formatted.text;
+            return [formatted.text];
         if (!Array.isArray(formatted))
             formatted = [formatted];
 
+        // filter out null/undefined in the array
+        let filtered = formatted.filter((formatted) => !!formatted);
```

Comment:
!! will also throw out the numbers 0 and NaN. Not sure if that's a problem, as both should be converted to strings, but something to keep in mind.